### ISSUE TITLE
Laravel 5.4 Support phpseclib/phpseclib dependencies

### ISSUE THREAD

**Issue by stefensuhat** [tymondesigns/jwt-auth#982]
Run `composer require tymon/jwt-auth` 

Return me:

Your requirements could not be resolved to an installable set of packages.

     Problem 1
    - Installation request for tymon/jwt-auth ^0.5.9 -> satisfiable by tymon/jwt-auth[0.5.9].
    - Conclusion: remove phpseclib/phpseclib 2.0.4
    - Conclusion: don't install phpseclib/phpseclib 2.0.4
    - tymon/jwt-auth 0.5.9 requires namshi/jose 5.0.* -> satisfiable by namshi/jose[5.0.0, 5.0.1, 5.0.2].
    - namshi/jose 5.0.0 requires phpseclib/phpseclib ~0.3 -> satisfiable by phpseclib/phpseclib[0.3.0, 0.3.1, 0.3.10, 0.3.5, 0.3.6, 0.3.7, 0.3.8, 0.3.9].
    - namshi/jose 5.0.1 requires phpseclib/phpseclib ~0.3 -> satisfiable by phpseclib/phpseclib[0.3.0, 0.3.1, 0.3.10, 0.3.5, 0.3.6, 0.3.7, 0.3.8, 0.3.9].
    - namshi/jose 5.0.2 requires phpseclib/phpseclib ~0.3 -> satisfiable by phpseclib/phpseclib[0.3.0, 0.3.1, 0.3.10, 0.3.5, 0.3.6, 0.3.7, 0.3.8, 0.3.9].
    - Can only install one of: phpseclib/phpseclib[0.3.0, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.1, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.10, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.5, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.6, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.7, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.8, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.9, 2.0.4].
    - Installation request for phpseclib/phpseclib (locked at 2.0.4) -> satisfiable by phpseclib/phpseclib[2.0.4].

    Installation failed, reverting ./composer.json to its original content.

I've tried run the latest beta version return me:

Your requirements could not be resolved to an installable set of packages.

     Problem 1
    - Installation request for tymon/jwt-auth 1.0.0.beta-1 -> satisfiable by tymon/jwt-auth[1.0.0-beta.1].
    - Conclusion: remove laravel/framework v5.4.6
    - Conclusion: don't install laravel/framework v5.4.6
    - tymon/jwt-auth 1.0.0-beta.1 requires illuminate/auth 5.1.* || 5.2.* || 5.3.* -> satisfiable by illuminate/auth[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.28, v5.1.30, v5.1.31, v5.1.41, v5.1.6, v5.1.8, v5.2.0, v5.2.19, v5.2.21, v5.2.24, v5.2.25, v5.2.26, v5.2.27, v5.2.28, v5.2.31, v5.2.32, v5.2.37, v5.2.43, v5.2.45, v5.2.6, v5.2.7, v5.3.0, v5.3.16, v5.3.23, v5.3.4].
    - don't install illuminate/auth v5.3.0|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.3.16|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.3.23|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.3.4|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.1|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.13|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.16|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.2|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.20|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.22|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.25|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.28|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.30|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.31|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.41|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.6|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.1.8|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.0|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.19|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.21|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.24|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.25|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.26|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.27|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.28|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.31|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.32|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.37|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.43|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.45|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.6|don't install laravel/framework v5.4.6
    - don't install illuminate/auth v5.2.7|don't install laravel/framework v5.4.6
    - Installation request for laravel/framework (locked at v5.4.6, required as 5.4.*) -> satisfiable by laravel/framework[v5.4.6].

**Comment by WanderingKazuma** [comment]
I am seeing the same:
```

Problem 1
    - Installation request for tymon/jwt-auth ^0.5.10 -> satisfiable by tymon/jwt-auth[0.5.10].
    - Conclusion: remove phpseclib/phpseclib 2.0.4
    - Conclusion: don't install phpseclib/phpseclib 2.0.4
    - tymon/jwt-auth 0.5.10 requires namshi/jose 5.0.* -> satisfiable by namshi/jose[5.0.0, 5.0.1, 5.0.2].
    - namshi/jose 5.0.0 requires phpseclib/phpseclib ~0.3 -> satisfiable by phpseclib/phpseclib[0.3.0, 0.3.1, 0.3.10, 0.3.5, 0.3.6, 0.3.7, 0.3.8, 0.3.9].
    - namshi/jose 5.0.1 requires phpseclib/phpseclib ~0.3 -> satisfiable by phpseclib/phpseclib[0.3.0, 0.3.1, 0.3.10, 0.3.5, 0.3.6, 0.3.7, 0.3.8, 0.3.9].
    - namshi/jose 5.0.2 requires phpseclib/phpseclib ~0.3 -> satisfiable by phpseclib/phpseclib[0.3.0, 0.3.1, 0.3.10, 0.3.5, 0.3.6, 0.3.7, 0.3.8, 0.3.9].
    - Can only install one of: phpseclib/phpseclib[0.3.0, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.1, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.10, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.5, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.6, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.7, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.8, 2.0.4].
    - Can only install one of: phpseclib/phpseclib[0.3.9, 2.0.4].
    - Installation request for phpseclib/phpseclib (locked at 2.0.4) -> satisfiable by phpseclib/phpseclib[2.0.4].
```

**Comment by ssuhat** [comment]
beta-1 fix this.

**Comment by garhbod** [comment]
Is the transition from ^0.5 to 1.0.0.beta easy?
Is there any migration required?

**Comment by tymondesigns** [comment]
fixed via `0.5.12`
